### DISCUSSION TITLE
Pin python-matplotlib to python version specified in depends_on

### DISCUSSION THREAD

**Discussion by kpeeters** [Homebrew/discussions#5824] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

I have a homebrew formula with `depends_on "python@3.12"`. I also have a `depends_on "python-matplotlib"` line, but that installs matplotlib for python3.13. How can I pin that to 3.12 too?

(I know I can install the lot in a venv by installing matplotlib using pip, but that's quite some additional code pulled in for something that is likely to be already present. Nothing in my formula depends critically on any particular version of matplotlib, so as long as it is present on the system the package will run).

**Comment by SMillerDev** [comment, marked as answer]
> How can I pin that to 3.12 too?

There is no way to do that. Unfortunately `python-matplotlib` isn't build against multiple python versions.
